### PR TITLE
feat: add tag

### PR DESCRIPTION
**Pull request by tanvibhakta**
## Description

Zeplin: https://zpl.io/DlrA5kv
Finishes https://github.com/appsmithorg/appsmith/issues/19535

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?
- Manual on storybook 
- Manual on main repo
- Jest
- Cypress

### Issues raised during DP testing
https://github.com/appsmithorg/design-system/pull/352#issuecomment-1425708034
https://github.com/appsmithorg/design-system/pull/352#issuecomment-1426984673

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: packages/design-system/src/Tag/Tag.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  title: "Design System/Tag",
+  component: Tag,
+} as ComponentMeta<typeof Tag>;
+
+// eslint-disable-next-line react/function-component-definition
+const Template: ComponentStory<typeof Tag> = (args) => {
+  return <Tag {...args} />;
+};
+
+export const TagExample = Template.bind({});
```
Comment (albinAppsmith) [inline packages/design-system/src/Tag/Tag.stories.tsx:16]: TagStory

File: packages/design-system/src/Tag/Tag.types.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+
+export type Size = "small" | "medium";
```
Comment (albinAppsmith) [inline packages/design-system/src/Tag/Tag.types.tsx:3]: this should be sm, md. Please use Sizes constant from __config__/types.
export type TagSizes = Extract<Sizes, "sm" | "md">

**Review by albinAppsmith — APPROVED** [review]

**Comment by shadabbuchh** [comment]
Bugs in this PR:
**Design:**

1. The medium size has a padding of 2 4px whereas expected is 0 4px.

**Comment by shadabbuchh** [comment]
Additional issues found:

1. Incorrect (2px) Gap between the `x` icon & the text.
2. The `x` icon is not appropriately aligned with the text.
Screenshot 2023-02-12 at 2 57 21 PM

**Comment by tanvibhakta** [comment]
The padding you can see around "x" is handled by the parent container and the icon container.

The alignment is set correctly. It might look visually different because of the different labels used. 

Screenshot 2023-02-14 at 9 26 38 AM

**Comment by shadabbuchh** [comment]
All the issues have been fixed and the correct specs have been implemented. However, @tanvibhakta I still don't believe that the `x` icon is appropriately aligned with the rest of the text (despite implementing it per design specs. Also making it look smaller).
Screenshot 2023-02-14 at 10 45 56 AM

Just highlighting this issue because anyone looking at this component in the future will be able to make this out with even inspecting the component.

**Pull request merged**